### ISSUE TITLE
ShareX not recording audio - I have installed all required recording devices and have no errors?

### ISSUE THREAD

**Issue by farazk86** [ShareX/ShareX#4730]
Hi,

Previously I had an old i3 desktop and was using shareX without any problems. Now I got a new PC build and reinstalled all of my software on the new PC. But now I get the screen to record but am not getting any sound recorded. 

![12](https://user-images.githubusercontent.com/33456896/78727549-d9ee0c80-792c-11ea-8f17-3d936356abff.PNG)

Here is the output of the cmd test:

```
ffmpeg version git-2020-04-03-52523b6 Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 9.3.1 (GCC) 20200328
  configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libdav1d --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libsrt --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-ffnvcodec --enable-cuda-llvm --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt --enable-amf
  libavutil      56. 42.102 / 56. 42.102
  libavcodec     58. 77.101 / 58. 77.101
  libavformat    58. 42.100 / 58. 42.100
  libavdevice    58.  9.103 / 58.  9.103
  libavfilter     7. 77.101 /  7. 77.101
  libswscale      5.  6.101 /  5.  6.101
  libswresample   3.  6.100 /  3.  6.100
  libpostproc    55.  6.100 / 55.  6.100
leaving aero onGuessed Channel Layout for Input Stream #0.1 : stereo
Input #0, dshow, from 'video=screen-capture-recorder:audio=virtual-audio-capturer':
  Duration: N/A, start: 168457.919000, bitrate: 1536 kb/s
    Stream #0:0: Video: rawvideo, bgr0, 1920x1080, 30 fps, 30 tbr, 10000k tbn, 10000k tbc
    Stream #0:1: Audio: pcm_s16le, 48000 Hz, stereo, s16, 1536 kb/s
Stream mapping:
  Stream #0:0 -> #0:0 (rawvideo (native) -> h264 (libx264))
  Stream #0:1 -> #0:1 (pcm_s16le (native) -> vorbis (libvorbis))
Press [q] to stop, [?] for help
[libx264 @ 0000015e903bb280] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
[libx264 @ 0000015e903bb280] profile Constrained Baseline, level 4.0, 4:2:0, 8-bit
[libx264 @ 0000015e903bb280] 264 - core 159 - H.264/MPEG-4 AVC codec - Copyleft 2003-2019 - http://www.videolan.org/x264.html - options: cabac=0 ref=1 deblock=0:0:0 analyse=0:0 me=dia subme=0 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=16 chroma_me=1 trellis=0 8x8dct=0 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=4 lookahead_threads=4 sliced_threads=1 slices=4 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=0 weightp=0 keyint=250 keyint_min=25 scenecut=0 intra_refresh=0 rc=crf mbtree=0 crf=28.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=0
Output #0, mp4, to 'output.mp4':
  Metadata:
    encoder         : Lavf58.42.100
    Stream #0:0: Video: h264 (libx264) (avc1 / 0x31637661), yuv420p, 1920x1080, q=-1--1, 30 fps, 15360 tbn, 30 tbc
    Metadata:
      encoder         : Lavc58.77.101 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/0 buffer size: 0 vbv_delay: N/A
    Stream #0:1: Audio: vorbis (libvorbis) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp
    Metadata:
      encoder         : Lavc58.77.101 libvorbis
[mp4 @ 0000015e903b4c80] Starting second pass: moving the moov atom to the beginning of the filespeed=1.01x
frame=  207 fps= 30 q=18.0 Lsize=    1403kB time=00:00:06.89 bitrate=1668.1kbits/s dup=62 drop=0 speed=   1x
video:1311kB audio:79kB subtitle:0kB other streams:0kB global headers:4kB muxing overhead: 0.959187%
```

I am not getting any errors. How do I record sound ? Ive tried all 3 audio codecs as well?

Thanks

**Comment by rescenic** [comment]
Install audio driver first from manufacture website.

**Comment by farazk86** [comment]
> Install audio driver first from manufacture website.

I have that installed, and my audio is working without problems in other apps

**Comment by r2dr0dn** [comment]
Hi @farazk86 did you managed to solve the problem , i'm having the same issue i as well have installed the latest audio drivers and everything works fine, but when recording using audio nothing even starts and no audio.

**Comment by farazk86** [comment]
Hi @r2dr0dn 

Yes, I managed to fix it in a strange way. I downloaded latest drivers from manufacturer (realtek, not dell) website. But that cause my back panel audio port to stop working. However the front headphone panel works. So I connected my speaker to the front and now I have sound and ShareX is also recording sound.

Its very strange and may not work for you, but this is what worked for me. It was a driver issue after all.
